discord/guild_const.go

```diff
@@ -49,38 +53,85 @@ const (
 	Banner GuildFeature = "BANNER"
 )
 
-type ExplicitFilter uint8
+// ExplicitFilter is the explicit content filter level of a guild.
+type ExplicitFilter nullable.Enum
 
-const (
-	NoContentFilter ExplicitFilter = iota
-	MembersWithoutRoles
-	AllMembers
+var (
+	// NullExplicitFilter serialized to JSON null.
+	// This should only be used on nullable fields.
+	NullExplicitFilter ExplicitFilter = nullable.EnumNull
+	// NoContentFilter disables content filtering for the guild.
+	NoContentFilter ExplicitFilter = 0
+	// MembersWithoutRoles filters only members without roles.
+	MembersWithoutRoles ExplicitFilter = 1
+	// AllMembers enables content filtering for all members.
+	AllMembers ExplicitFilter = 2
 )
 
-type Notification uint8
+func (f *ExplicitFilter) UnmarshalJSON(b []byte) error {
```

Comment:
(If we're keeping this), `ExplicitFilter` should've already inherited `UnmarshalJSON` from `nullable.Enum` right?

Reply:
Thought that as well, but unfortunately no. Inheritance seems to be the only thing for structs but not for type wrapping, so we have to call the function ourselves.
And as we can't use them directly, I decided to simplify things for the caller and don't implement the `Unmarshaler` and `Marshaler` interfaces on them, but to make a function for this, as this shortens UnmarshalJSON and MarshalJSON.